### DISCUSSION TITLE
Automatic cleanup - yay or nay?

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#32] [General]
So nextflow now gives us the option to perform an automatic cleanup of the work directory upon successful pipeline completion (basically just needs "cleanup = true" in nextflow.config). 

I am leaning towards adding this, because I have found in the past that people usually forget to run this manually and then complain about their storage quota being used up at some point. 

But - this would disable the possibility of resuming a run without re-computing everything. From a developer point of view this could be a little inconvenient.

**Comment by marchoeppner** [comment]
A compromise would be to not add this to the default config, but instead to each site-specific config - if the users is OK with it (i.e. the configs hosted here: https://github.com/marchoeppner/nf-configs

**Comment by gregdenay** [comment]
A big Yay from me. Average user will be running each analysis only once.

The only use cases I can see where you would want to rerun several times would be parameter optimization and debugging. As this would mostly be done by "advanced" users I think it would be reasonnable to mention the parameter in the doc and letting users chose to keep the work diretory alive on purpose.
